### PR TITLE
Support classes with type parameters and @Inject constructor

### LINKED ISSUE #111: Support for @Inject with generic types
Currently processing:
```
class InjectClass<T> @Inject constructor(val provider: Provider<T>)
```
is valid in Dagger but throws an error in Anvil

### PR DESCRIPTION
**Pull request by quiro91**
Fixes #111 
Feedbacks very appreciated, I don't have much experience with Kotlin psi 😅

### CODE REVIEW

**Comment by quiro91** [comment]
Instrumentation tests got stuck maybe because of the emulator?

**Comment by vRallev** [comment]
Sorry, at the moment I don't have a chance to take a look. I'll get back to your PR during the next days.

**Comment by quiro91** [comment]
No rush @vRallev, whenever you can

File: compiler/src/main/java/com/squareup/anvil/compiler/codegen/KotlinPoet.kt [hunk 1]
```diff
@@ -97,7 +98,13 @@ internal fun KtTypeReference.requireTypeName(
   fun KtTypeElement.requireTypeName(): TypeName {
     return when (this) {
       is KtUserType -> {
-        val className = requireFqName(module).asClassName(module)
+        val className = fqNameOrNull(module)?.asClassName(module)
+            ?: if (isTypeParameter()) {
```
Comment (vRallev) [inline compiler/src/main/java/com/squareup/anvil/compiler/codegen/KotlinPoet.kt:102]: I've added a check here.

**Review by vRallev — APPROVED** [review]
This PR is great. I've added an additional check and corrected some of the style changes. Thanks for the PR!

**Comment by quiro91** [comment]
Thanks for the improvements!

**Pull request merged**